pkg/lib/operation/project/remote/job/run/operation.go

```diff
@@ -0,0 +1,220 @@
+package run
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/options"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type dependencies interface {
+	KeboolaProjectAPI() *keboola.API
+	Logger() log.Logger
+	Tracer() trace.Tracer
+}
+
+func Run(ctx context.Context, o RunOptions, d dependencies) (err error) {
+	ctx, span := d.Tracer().Start(ctx, "kac.lib.operation.project.remote.job.run")
+	defer telemetry.EndSpan(span, &err)
+
+	logger := d.Logger()
+	api := d.KeboolaProjectAPI()
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, o.Timeout)
+	defer cancel()
+
+	// remaining stores keys of jobs which are still running
+	remaining := map[string]bool{}
+	remainingMutex := &sync.Mutex{}
+	jobErrors := errors.NewMultiError()
+	wg := &sync.WaitGroup{}
+
+	if len(o.Jobs) > 1 {
+		logger.Infof("Starting %d jobs", len(o.Jobs))
+	} else {
+		logger.Info("Starting 1 job")
+	}
+
+	// dispatch jobs in parallel
+	for _, job := range o.Jobs {
+		job := job
+
+		remaining[job.Key] = true
+		wg.Add(1)
+		go dispatchJob(timeoutCtx, api, &job, wg, jobErrors, remaining, remainingMutex)
```

Comment:
Odovzdavas tu dost vela argumentov, bolo by lepsie, ak by si si definoval nejaku strukturu, neviem napr. `JobsQueue{wg,hasQeue2,remaining,lock,....}` a tam mal tie zavislosti + mohlo by to byt v samostatnom subore a bolo by to aj testovatelne.

A nasledujuce funkcie by boli metody.